packages/core/src/crawlers/crawler_commons.ts

```diff
@@ -81,3 +81,17 @@ export interface CrawlingContext<Crawler = unknown, UserData extends Dictionary
      */
     sendRequest<Response = string>(overrideOptions?: Partial<OptionsInit>): Promise<GotResponse<Response>>;
 }
+
+export interface RestrictedCrawlingContext<UserData extends Dictionary = Dictionary> {
+    request: Request<UserData>;
+    pushData: (record: Dictionary, datasetName?: string) => Promise<void>;
+    enqueueLinks: (options?: EnqueueLinksOptions) => Promise<void>;
+    addRequests: (
+        requestsLike: Source[],
+        options?: RequestQueueOperationOptions,
+    ) => Promise<void>;
+    getStoreValue: <Value = unknown>(key: string, defaultValue?: Value) => Promise<Value>
+    setStoreValue: <Value = unknown>(key: string, value: Value) => Promise<void>
```

Comment:
This is missing the third optional `options` param, is this intentional? How do we pass the `content-type`?

Reply:
oversight on my part - thanks